app/views/retros/new.html.erb

```diff
@@ -1,11 +1,16 @@
 <% content_for :head do %>
   <title>New retro - WyeRetroApp</title>
 <% end %>
-<header id="header">
-  <div class='nav'>
-    <%= image_tag("/assets/wye-logo.png", size: "40x40", class: "header-icon") %>
+<div id="header">
+  <div id="nav">
+    <div class="headerIcons">
+      <Button id="start-new-button" class="start-button">Post your topics!</Button>
+      <Button id="time"></Button>
+      <%= image_tag("/assets/wye-logo.png", size: "40x40", class: "header-icon") %>
+    </div>
   </div>
-</header>
-<div id="buttons">
-  <%= button_to "Start new Retro!", '/retros/new', method: :get, class: "start-button" %>
 </div>
+
+<% content_for :script do %>
+  <%= javascript_include_tag 'retros' %>
```

Comment:
The reason why I removed the `javascript_include_tag` in `application.html.erb` is because the following error was being displayed in the console when loading the main page:

<img width="573" alt="screen shot 2018-10-17 at 15 50 59" src="https://user-images.githubusercontent.com/8642789/47109400-8366ad00-d224-11e8-8191-188b87dbe38f.png">

This is because as you said, all the .js files are included in every page, therefore `retros.js` looks for elements that belong to `retros/new` in `/` and doesn't find them, so they are undefined even though the document is indeed ready.

Maybe there's another way to fix this issue ?

Reply:
I think that the problem there is that if we are not in `/retros/new` then `startNewRetroButton` in `retros.js` is undefined. So maybe this error can be fixed asking if this variable is defined before adding an event listener to it.

Anyway, I think that there has to be a better way to organize the javascript code without being defensive and hacky. Maybe Rodri or Larry knows more about that

Reply:
This will be solved in https://github.com/anita040593/RetroApp/pull/6 because I'll add `<%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>` again to `application.html.erb` without having to worry about elements not being in the document by using the JQuery selector:

> Calling jQuery() (or $()) with an id selector as its argument will return a jQuery object containing a collection of either zero or one DOM element.
[-> Source](https://api.jquery.com/id-selector/)

It won't try to add an event listener to null (which is returned by .getElementById when the element is not found) and when .on() is called upon the object returned from $('#id') with zero DOM elements it tries to apply to the handler to each object in the collection, since there are no objects it "doesn't do" anything and continues on with the code, no error messages are shown in the console and we can include application.js directly :D

This was also useful to read:
http://api.jquery.com/on/
http://api.jquery.com/Types/#PlainObject
https://stackoverflow.com/questions/5724751/clarification-of-jquery-wrapper-object

Reply:
That's much better! And jQuery will be useful for other things too! 💪